### PR TITLE
Simplify shallow resolver to just fold ty/consts

### PR DESCRIPTION
**Pull request by compiler-errors**
Probably faster than using a whole folder?

### CODE REVIEW

File: compiler/rustc_infer/src/infer/mod.rs [hunk 1]
```diff
@@ -1245,19 +1245,67 @@ impl<'tcx> InferCtxt<'tcx> {
...
+    pub fn shallow_resolve_const(&self, ct: ty::Const<'tcx>) -> ty::Const<'tcx> {
+        match ct.kind() {
+            ty::ConstKind::Infer(InferConst::Var(vid)) => self
+                .inner
+                .borrow_mut()
+                .const_unification_table()
+                .probe_value(vid)
+                .known()
+                .unwrap_or(ct),
+            ty::ConstKind::Infer(InferConst::EffectVar(vid)) => self
```
Comment (lcnr) [inline compiler/rustc_infer/src/infer/mod.rs:1300]: exhaustive match on `ty::ConstKind::Infer` please

**Review by lcnr — COMMENTED** [review]
r=me if perf's good

**Comment by Kobzol** [comment]
The CI failure is unrelated to this PR, should be fixed by https://github.com/rust-lang/rust/pull/123963.

**Pull request merged**